src/GraphQL/Types/Scalars/DecimalGraphType.cs

```diff
@@ -10,40 +10,22 @@ namespace GraphQL.Types
     public class DecimalGraphType : ScalarGraphType
     {
         /// <inheritdoc/>
-        public override object? ParseLiteral(IValue value) => value switch
+        public override object? ParseLiteral(GraphQLValue value) => value switch
         {
-            IntValue intVal => (decimal)intVal.Value,
-            LongValue longVal => (decimal)longVal.Value,
-            FloatValue floatVal => checked((decimal)floatVal.Value),
-            DecimalValue decVal => decVal.Value,
-            BigIntValue bigIntVal => checked((decimal)bigIntVal.Value),
-            NullValue _ => null,
+            GraphQLIntValue x => Decimal.Parse(x.Value),
+            GraphQLFloatValue x => Decimal.Parse(x.Value),
```

Comment:
This is so much better than having a bunch of conditions try to find the best conversion, only to have the scalar convert the value to the desired type.

Reply:
Yes. Only ParseValue has conversions.